resource-sharing-partners/src/test/java/no/sikt/rsp/ResourceSharingPartnerTest.java

```diff
@@ -851,6 +855,126 @@ public void shouldPopulateLocateProfileInPartnerDetailsCorrectly(final String ka
+
+        final TestAppender appender = LogUtils.getTestingAppenderForRootLogger();
+
+        final Integer count = resourceSharingPartnerHandler.handleRequest(s3Event, CONTEXT);
+
+        assertThat(count, is(0));
+        assertThat(appender.getMessages(), containsString(LOG_MESSAGE_COMMUNICATION_PROBLEM));
+    }
+
+    @Test
+    public void shouldIgnorePartnerAndLogProblemIfAlmaIsNotAvailable() throws IOException {
+        var uri = s3Driver.insertFile(randomS3Path(), BIBNR_RESOLVABLE_TO_ALMA_CODE);
+        var s3Event = createS3Event(uri);
+
+        final Record record = new RecordBuilder(BigInteger.ONE, LocalDate.now(), "BIBSYS").withBibnr(
+            BIBNR_RESOLVABLE_TO_ALMA_CODE).withLandkode("1").build();
+
+        var basebibliotekGenerator = new BasebibliotekGenerator(record);
+        var basebibliotek = basebibliotekGenerator.generateBaseBibliotek();
+        var basebibliotekXml = BasebibliotekGenerator.toXml(basebibliotek);
+
+        WireMocker.mockBasebibliotekXml(basebibliotekXml, BIBNR_RESOLVABLE_TO_ALMA_CODE);
+
+        when(mockedEnvironment.readEnv(ResourceSharingPartnerHandler.ALMA_API_HOST)).thenReturn(
+            UriWrapper.fromUri("http://localhost:9999").toString());
+
+        resourceSharingPartnerHandler = new ResourceSharingPartnerHandler(s3Client, mockedEnvironment);
+
+        final TestAppender appender = LogUtils.getTestingAppenderForRootLogger();
+
+        final Integer count = resourceSharingPartnerHandler.handleRequest(s3Event, CONTEXT);
+
+        assertThat(count, is(0));
+        assertThat(appender.getMessages(),
+                   containsString(HttpUrlConnectionAlmaPartnerUpdater.LOG_MESSAGE_COMMUNICATION_PROBLEM));
+    }
+
+    @Test
+    public void shouldIgnorePartnerAndLogProblemIfAlmaGetPartnerReturnsBadRequestWithUnhandledErrorCode() throws IOException {
+        var uri = s3Driver.insertFile(randomS3Path(), BIBNR_RESOLVABLE_TO_ALMA_CODE);
+        var s3Event = createS3Event(uri);
+
+        final Record record = new RecordBuilder(BigInteger.ONE, LocalDate.now(), "BIBSYS").withBibnr(
+            BIBNR_RESOLVABLE_TO_ALMA_CODE).withLandkode("NO").build();
+
+        var basebibliotekGenerator = new BasebibliotekGenerator(record);
+        var basebibliotek = basebibliotekGenerator.generateBaseBibliotek();
+        var basebibliotekXml = BasebibliotekGenerator.toXml(basebibliotek);
+
+        WireMocker.mockBasebibliotekXml(basebibliotekXml, BIBNR_RESOLVABLE_TO_ALMA_CODE);
+        WireMocker.mockAlmaGetResponseBadRequestNotPartnerNotFound(NO_0030100_ID);
+
+        final TestAppender appender = LogUtils.getTestingAppenderForRootLogger();
+
+        final Integer count = resourceSharingPartnerHandler.handleRequest(s3Event, CONTEXT);
+
+        assertThat(count, is(0));
+        assertThat(appender.getMessages(),
+                   containsString("Unexpected error response getting partner"));
```

Comment:
Her kunne du kanskje bruke direkte text fra AlmaError klassen (HttpUrlConnectionAlmaPartnerUpdater). Hvis vi komme en fin dag å endre teksten på errormessage må vi endre to plasser for at testen kjører.